contrib/incident-response/infrastructure/modules/turbinia/main.tf

```diff
@@ -0,0 +1,259 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Random ID for creating unique resource names.
+# TODO: Allow overide by config
+resource "random_id" "infrastructure-random-id" {
+  byte_length = 8
+}
+
+# Enable GCP services individually.
+resource "google_project_service" "cloudfunctions" {
+  project = "${var.gcp_project}"
+  service = "cloudfunctions.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "compute" {
+  project = "${var.gcp_project}"
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "datastore" {
+  project = "${var.gcp_project}"
+  service = "datastore.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "iam" {
+  project = "${var.gcp_project}"
+  service = "iam.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "pubsub" {
+  project = "${var.gcp_project}"
+  service = "pubsub.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "storage-component" {
+  project = "${var.gcp_project}"
+  service = "storage-component.googleapis.com"
+  disable_on_destroy = false
+}
+
+# Enable PubSub and create topic
+resource "google_pubsub_topic" "pubsub-topic" {
+  name = "turbinia-${random_id.infrastructure-random-id.hex}"
+  depends_on  = ["google_project_service.pubsub"]
+}
+
+resource "google_pubsub_topic" "pubsub-topic-psq" {
+  name        = "turbinia-${random_id.infrastructure-random-id.hex}-psq"
+  depends_on  = ["google_project_service.pubsub"]
+}
+
+# Cloud Storage Bucket
+resource "google_storage_bucket" "output-bucket" {
+  name          = "turbinia-${random_id.infrastructure-random-id.hex}"
+  depends_on    = ["google_project_service.storage-component"]
+  force_destroy = true
+}
+
+# Create datastore index
+data "local_file" "datastore-index-file" {
+  filename = "${path.module}/data/index.yaml"
+  depends_on  = ["google_project_service.datastore"]
+}
+
+resource "null_resource" "cloud-datastore-create-index" {
+  provisioner "local-exec" {
+    command = "gcloud -q datastore indexes create ${data.local_file.datastore-index-file.filename} --project=${var.gcp_project}"
+  }
+}
+
+# Deploy cloud functions
+data "archive_file" "cloudfunction-archive" {
+  type        = "zip"
+  output_path = "${path.module}/data/function.zip"
+
+  source {
+    content  = "${file("${path.module}/data/function.js")}"
+    filename = "function.js"
+  }
+
+  source {
+    content  = "${file("${path.module}/data/package.json")}"
+    filename = "package.json"
+  }
+}
+
+resource "google_storage_bucket_object" "cloudfunction-archive" {
+  name   = "function.zip"
+  bucket = "${google_storage_bucket.output-bucket.name}"
+  source = "${path.module}/data/function.zip"
+  depends_on = ["data.archive_file.cloudfunction-archive"]
+}
+
+resource "google_cloudfunctions_function" "gettasks" {
```

Comment:
Would it work to consolidate each of the cloudfunctions into a list in `locals` and use a count of the list to create each individual func? The params look like they are shared cross all three- if these can differ in the future/in another use case, ok to just leave as is.

Reply:
Done.